docs/architecture/adr-008-blocksize-independent-commitment.md

```diff
@@ -0,0 +1,100 @@
+
+TODO
+
+## Decision
+
+TODO
+
+## Detailed Design
+
+To implement this decision, you need to change [`CreateCommit`](https://github.com/celestiaorg/celestia-app/blob/0c81704939cd743937aac2859f3cb5ae6368f174/x/payment/types/payfordata.go#L112-166).
+In Detail, [`powerOf2MountainRange`](https://github.com/celestiaorg/celestia-app/blob/0c81704939cd743937aac2859f3cb5ae6368f174/x/payment/types/payfordata.go#L142) should take `minimalBlocksize` as an argument instead of `squareSize`.
+
+Following the non-interactive default rules, `minimalBlocksize` can be calculated like this:
+
+```go
+func minimalBlocksize(messageLength uint64) uint64 {
+    blocksize := NextHigherPowerOf2(messageLength)
+    //Check if message fits with non-interactive default rules
+    if messageLength <= (blocksize * (blocksize -1)) {
+        return blocksize
+    } else {
+        return blocksize << 1
+    }
+}
+```
+
+## Status
+
+Proposed
+
+## Consequences
+
+### Negative
+
+1. The amount of subtree roots per commitment is O(sqrt(n)), while n is the number of message shares. The worst case for the number of subtree roots is depicted in the diagram below - an entire block missing one share.  
+  ![Interactive Commitment 2](./assets/complexity.png)  
+The worst case for the current implementation depends on the block size. If it is the worst block size, as in `minimalBlocksize`, it is O(sqrt(n)) as well. On the other hand, if the message is only in one row, then it is O(log(n)).
+Therefore the height of the tree over the subtree roots is in this implementation O(log(sqrt(n))), while n is the number of message shares. In the current implementation, it varies from O(log(sqrt(n))) to O(log(log(n))) depending on the block size.
+
+2. With more subtree roots, the number of merkle proofs will increase. With deeper subtree roots, the size of the merkle proofs will increase. So instead of having a merkle proof from the `DataRoot` to the `originalSubRoot` you will need the merkle proof from `DataRoot` to `2^k` amount of `miniSubRoots` with `k` being the height difference of `originalSubRoot` and `miniSubRoots`. You can optimize this by having only **one** merkle proof from `DataRoot` to `originalSubRoot` and then calculate the `originalSubRoot` from the `k` `miniSubRoots`. Because the merkle proof is created after the block is published, we know the block size and, therefore, if this `originalSubRoot` exists or not.
+
+### Positive
+
+1. A Rollup can include the commitment in the block header *before* posting to Celestia because it is size-independent and does not have to wait for Celestia to confirm the block size. In general, the rollup needs access to this commitment in some form to verify a message inclusion proof guaranteeing data availability, which Rollmint currently does not have access to.
+2. In turn, this would serve as an alternative to [ADR 007](https://github.com/celestiaorg/optimint/blob/main/docs/lazy-adr/adr-007-header-commit-to-shares.md)
```

Comment:
[optional] to disambiguate with adr-007 in this repo

```suggestion
2. In turn, this would serve as an alternative to [rollmint/adr-007](https://github.com/celestiaorg/optimint/blob/main/docs/lazy-adr/adr-007-header-commit-to-shares.md)
```